app/controllers/api/automate_workspaces_controller.rb

```diff
@@ -4,5 +4,32 @@ def edit_resource(type, id, data = {})
       obj = resource_search(id, type, collection_class(type))
       obj.merge_output!(data)
     end
+
+    def decrypt_resource(type, id = nil, data = nil)
+      raise BadRequestError, "Must specify an id for starting a #{type} resource" unless id
+
+      obj = resource_search(id, type, collection_class(type))
+      decrypt(obj, data)
+    end
+
+    def encrypt_resource(type, id = nil, data = nil)
+      raise BadRequestError, "Must specify an id for starting a #{type} resource" unless id
+
+      obj = resource_search(id, type, collection_class(type))
+      obj.encrypt(data['object'], data['attribute'], data['value'])
+      obj.reload
+      obj
+    end
+
+    private
+
+    def decrypt(obj, data)
+      begin
+        value = obj.decrypt(data['object'], data['attribute'])
+      rescue
+        value = ""
```

Comment:
Could you add a test for this case where we rescue the error? What kind of error are you anticipating? This seems like the sort of thing that `obj` here should handle internally

Reply:
@imtayadeway Catching and returning an empty string was suggested by @Fryguy to thwart Oracle attack, trying to decipher the encryption key, which has a very remote possibility here.
The app/models code raises the exception when the object, attribute is missing or when the attribute is not a password type.

Reply:
Maybe you could extract a `#decrypt!` method?

```ruby
def decrypt!(object, attribute)
  # whatever old #decrypt does
end

def decrypt(object, attribute)
  decrypt!(object, attribute)
rescue TheSpecificError
  ""
end
```

I don't think handling this in the controller is a good idea - it's a piece of knowledge about how to recover from such an event that needs to be encapsulated in the model.